src/pages/auth/login.vue

```diff
@@ -80,21 +105,41 @@ const loading = ref(false);
 const data = reactive<LoginData>({
   email: '',
   password: '',
+  token: '',
 });
 
 const hasCredentialError = ref(false);
+const hasMFAEnabled = ref(false);
+const hasWrongMFAToken = ref(false);
+
+let emailBefore = '';
 
 async function submitLogin() {
   loading.value = true;
   hasCredentialError.value = false;
+  hasWrongMFAToken.value = false;
+
+  // Only reset MFA if email changed
+  if (emailBefore !== data.email) {
+    hasMFAEnabled.value = false;
+  }
+  emailBefore = data.email;
   try {
     await login(data);
     await updateCurrentUser();
     // TODO: use router when legacy app is gone
     window.location.href = isInternalUrl(redirectTo) ? redirectTo : '/';
     // eslint-disable-next-line @typescript-eslint/no-explicit-any
   } catch (err: any) {
-    if (err.response?.status === 401) hasCredentialError.value = true;
+    if (err.response?.status === 401) {
```

Comment:
You could use `isApiError(err, code)` here, it handles checking the type so `err` can be `unknown` instead.

It currently only checks for 400 status code, but you could change the signature to
```ts
function isApiError(err: unknown, code: string, status?: number = 400)
```
and update accordingly.

Reply:
I can't find `isApiError`, do you mean `isRequestError`? But this method just checks for `err.response?.status === 400` should this be `err.response?.status >= 400`?

Reply:
Ah sorry I meant `isRequestError`. I think it would be better to add an optional third parameter to change the status code it checks for, e.g.
```ts
function isRequestError(err: unknown, code: string, status?: number  = 400) {
...
}
```